chat-server/src/integrations/openai.ts

```diff
@@ -45,3 +54,90 @@ export class OpenAiEmbeddingsClient implements OpenAiEmbeddingsClientInterface {
     return { status, data };
   }
 }
+
+interface OpenAiChatClientInterface {
```

Comment:
Probably as a rule: export interfaces (there's no harm in doing it and it makes it easier later when someone inevitably wants to use it)
```suggestion
export interface OpenAiChatClientInterface {
```

Philosophical nitpick: I actually think the naming convention should not include `Interface`. Instead, make the interface the generic name (`OpenAiChatClient`) and give the implementation the decoration (one example: `OpenAiChatClientImpl` or interface `ChatClient`, impl `OpenAiChatClient`). That way you remember to design your functions that take a chat client to take the interface, not the specific implementation!